# Najbolj popularni animeji
## Projektna naloga pri predmetu Programiranje 1.

Za to projektno nalogo sem zajel prvih 3000 najbolj popularnih animejev s strani [MyAnimeList](https://myanimelist.net/topanime.php). Podatki so rangirani glede na [tale](https://myanimelist.net/info.php?go=topanime) kriterij.
Za pridobitev in obdelavo podatkov sem, bolj kot ne, uporabil kodo, ki jo je profesor Pretnar pripravil za predavanja.

#### Naloga je sestavljena iz štirih delov:
1. ZBRANI PODATKI:
    - Definicija animeja
    - Delovne hipoteze
    - Z zajemom sestavljena tabela.
2. OSNOVNA STATISTIKA: Podatki, pridobljeni s preureditvijo tabele.
3. IZRAČUNANA STATISTIKA: Podatki, ki jih moramo izračunati oz. razbrati, odvisnosti in podobno.
4. DODATNE HIPOTEZE: za zabavo bom morda pogledal še kakšno hipotezo, ki je nisem napovedal v Readme.md datoteki.



### ZBRANI PODATKI
* Podatki ki sem jih zbral za posamezni anime:
    - rang
    - id
    - naslov
    - leto izida
    - dolžina(število epizod)
    - tip
    - MyAnimeList ocena
    - število ogledov

Glavni razlog, zakaj nisem zajel več različnih tipov podatkov, je dejstvo, da če bi želel zajeti več podatkov (npr. studio/ji, ki je/so sproduciral/i anime), bi moral za vsaki anime posebej dol potegniti spletno stran (torej 3000 spletnih strani). V prvi vrsti dvomim, da bi mi strežnik to dopustil v enem poskusu, kaj šele v vseh (mnogih) poskusih, ki sem jih izvedel, da sem sestavil to nalogo, v drugi vrsti pa je to res ogromna količina datotek, ki bi jih moral prečesati in v primeru napake, prečesati še enkrat. Krajše povedano: Časovna zahtevnost projekta bi se znatno povečala. 

#### Kaj je anime?
[Anime](https://en.wikipedia.org/wiki/Anime) je originalno japonski izraz za animacijo, torej vse oblike animiranih medijev (filmi, serije itd.). Izven Japonske se termin nanaša specifično na animacije, ki izvirajo iz Japonske, ali pa na animacijski, ki se je najprej razširil na Japonskem, za katerega je značilna tako vizualna kot tudi karakterna barvitost/pestrost in fantazijske teme. Spletna stran kot anime upošteva vse vzhodno-azijske animacije, ki so bodisi japonske, bodisi spadajo pod prej omenjeni stil.
Animeje se deli na več tipov, za nas so pomembni tile:
* TV: Animirana serija, ki je bila (najprej) predvajana po televiziji.
* Movie: Animiran film, ki je bil (najprej) predvajan v kinih.
* Special: Posebej izdana animacija, ki se jo lahko najde na televiziji, ali pa je izdana kot DVD.
* OVA: "O(riginal) V(ideo) A(nimation)", animacija, ki je direktno izdana na disku in predhodno ni predvajana na televiziji ali v kinih. 
* ONA: "O(riginal) N(etwork) A(nimation)", animacija, ki je najprej izdana na spletu.
O slednjih treh tipih si lahko več preberete [tukaj](https://aminoapps.com/c/anime/page/blog/defining-ova-ona-special-episodes-and-movies/bEto_u8PomgzmRaW7MLjQxz5LoZ6VP)

Kot sem prej omenil, je rang animeja določen s pomočjo kriterija s strani MyAnimeList. Tako bom na začetku malce pokomentiral ta kriterij.
Za izračun "Težnostnega ranga" animeja potrebujemo povprečno oceno tega animeja, število vseh ocen animeja, minimalno število ocen, da se sploh "splača" računat (trenutno je to 50 ocen) in povprečno oceno po vseh animejih. Anime lahko ocenijo samo registrirani uporabniki te spletne strani, kar pomeni, da podatki, ki so bili zbrani s te strani, niso nujno najbolj natančni, še posebej ne za starejše animeje.

Skozi vso projektno nalogo bom odgovarjal na sledeča vprašanja:
* Ali je zlata doba animejev že mimo, ali se šele začenja?
* Ali so bolj popularni animirani filmi, ali serije?
* Kako dolžina (TV)serije vpliva na njen uspeh?
* V kakšnem razmerju sta število ogledov in MyAnimeList ocena?

In [34]:
%matplotlib

import pandas as pd
import numpy as np
import matplotlib

from matplotlib import pyplot as plt

pd.options.display.max_rows = 20

animeji = pd.read_csv('obdelani-podatki/vsi-animeji.csv', index_col = 'rang')

Using matplotlib backend: TkAgg


In [35]:
print(matplotlib.__version__)

3.0.2


In [36]:
animeji

id                                             naslov      tip  \
rang                                                                      
1      5114                   Fullmetal Alchemist: Brotherhood       TV   
2     32281                                     Kimi no Na wa.    Movie   
3      9253                                        Steins;Gate       TV   
4     28977                                           Gintama°       TV   
5     11061                             Hunter x Hunter (2011)       TV   
6       820                               Ginga Eiyuu Densetsu      OVA   
7      9969                                           Gintama'       TV   
8     35180                         3-gatsu no Lion 2nd Season       TV   
9     15417                                Gintama': Enchousen       TV   
10    28851                                     Koe no Katachi    Movie   
...     ...                                                ...      ...   
2991    478                                 Sousei no Aquarion       TV   
2992   2452                              Space Adventure Cobra    Movie   
2993  35774                         Starmyu 2nd Season Special  Special   
2994   3332                                 Stop!! Hibari-kun!       TV   
2995  21881                                Sword Art Online II       TV   
2996   4532                          Taiyou no Yuusha Fighbird       TV   
2997  20035  Toaru Majutsu no Index Movie: Endymion no Kise...    Movie   
2998   7654                          Tokyo Magnitude 8.0 Recap  Special   
2999   2577                               Tom Sawyer no Bouken       TV   
3000  31370                               Tonkatsu DJ Agetarou       TV   

      st_epizod  leto   ogledi  ocena  
rang                                   
1            64  2009  1327051    9.2  
2             1  2016   866945    9.1  
3            24  2011  1116166    9.1  
4            51  2015   226008    9.1  
5           148  2011   818148    9.1  
6           110  1988   140128    9.1  
7            51  2011   235640    9.1  
8            22  2017   114343    9.0  
9            13  2012   137790    9.0  
10            1  2016   621825    9.0  
...         ...   ...      ...    ...  
2991         26  2005    57573    7.1  
2992          1  1982     5258    7.1  
2993          1  2017     1649    7.1  
2994         35  1983     2996    7.1  
2995         24  2014   911220    7.1  
2996         48  1991     1746    7.1  
2997          1  2013    15645    7.1  
2998          1  2009     5668    7.1  
2999         49  1980     3648    7.1  
3000         12  2016     9884    7.1  

[2930 rows x 7 columns]

Iz do sedaj neznanega razloga ni bilo zajetih 70 datotek. Zaradi tega av tabeli ni 3000 vrstic, temveč zgolj 2930. Napako bom poskusil odpraviti, če mi to ne uspe, pa je zajeti vzorec še vedno dovolj obsežen, da ta izguba ni občutna.

### OSNOVNA STATISTIKA
Oglejmo si, kateri izmed zajetih animejev je najstarejši(posledično bomo videli tudi kateri je najmlajši) in kateri je bil do zdaj najbolj gledan.

In [37]:
animeji_po_starosti = animeji.sort_values('leto')

In [38]:
animeji_po_starosti

id                                             naslov      tip  \
rang                                                                      
2847  33335            Hajimete no Eigo: Nippon Mukashibanashi      OVA   
2539   3624                     Seimei no Kagaku: Micro Patrol       TV   
2778   3186           Sekai Meisaku Douwa: Hakuchou no Mizuumi    Movie   
740    6591             Rurouni Kenshin DVD-BOX Special Ending  Special   
770    6154  Saint Seiya: The Hades Chapter Sanctuary - Yom...  Special   
984    7471            Detective Conan Movie 09: Promo Special    Movie   
2824   4162         Taiho Shichau zo: Nagisa no Koutsuu Yuudou  Special   
1237   6553                           Junjou Romantica Special  Special   
1280   1395                       Future GPX Cyber Formula Sin      OVA   
1381   2514  Detective Conan OVA 04: Conan and Kid and Crys...      OVA   
...     ...                                                ...      ...   
1719  37621                    Karakai Jouzu no Takagi-san OVA      OVA   
675   35222                                 Gakuen Babysitters       TV   
2856  38198          Nanatsu no Taizai: Eiyuu-tachi wa Hashagu      OVA   
249   34577            Nanatsu no Taizai: Imashime no Fukkatsu       TV   
2424  36511                                     Tokyo Ghoul:re       TV   
1285  37475                           Kishuku Gakkou no Juliet       TV   
1737  36653                  Tsurune: Kazemai Koukou Kyuudoubu       TV   
231   35608   Chuunibyou demo Koi ga Shitai! Movie: Take On Me    Movie   
1700  36432                         Toaru Majutsu no Index III       TV   
451   37989                            Golden Kamuy 2nd Season       TV   

      st_epizod  leto  ogledi  ocena  
rang                                  
2847         18     0   10223    7.2  
2539         26     1    3742    7.2  
2778          1     1    1623    7.2  
740           1     1  141748    7.8  
770           2     1   54645    7.8  
984           1     1   33298    7.7  
2824          1     1   50269    7.2  
1237          1     1   75894    7.6  
1280          5     1   33411    7.6  
1381          1     1    4623    7.5  
...         ...   ...     ...    ...  
1719          1  2018   25598    7.4  
675          12  2018   68454    7.9  
2856          1  2018   18907    7.2  
249          24  2018  314325    8.3  
2424         12  2018  316587    7.3  
1285         12  2018   81223    7.6  
1737         13  2018   34653    7.4  
231           1  2018   67819    8.3  
1700         26  2018   82365    7.4  
451          12  2018   35683    8.0  

[2930 rows x 7 columns]

Iz urejene tabele vidimo, da je najstarejši zajeti anime animiran film "Acchi Kocchi" iz leta 1962, najmlajši anime pa ONA "Isekai Izakaya: Koto Aitheria no Izakaya Nobu", ki je izšel leta 2018. Iz povezave v definiciji animeja pa vemo, da je prvi izšel že leta 1917, kar potrjuje, da je naš vzorec nepopoln.

In [39]:
animeji_po_ogledih = animeji.sort_values('ogledi', ascending=False)

In [40]:
animeji_po_ogledih

id                              naslov      tip  st_epizod  leto  \
rang                                                                        
53     1535                          Death Note       TV         37  2006   
113   16498                  Shingeki no Kyojin       TV         25  2013   
1410  11757                    Sword Art Online       TV         25  2012   
1      5114    Fullmetal Alchemist: Brotherhood       TV         64  2009   
45    30276                       One Punch Man       TV         12  2015   
581   22319                         Tokyo Ghoul       TV         12  2014   
3      9253                         Steins;Gate       TV         24  2011   
253    6547                        Angel Beats!       TV         13  2010   
177   19815                     No Game No Life       TV         12  2014   
716      20                              Naruto       TV        220  2002   
...     ...                                 ...      ...        ...   ...   
1941   9340                         Cheburashka    Movie          1  2010   
2963   9200  Dae Jang Geum: Jang Geum's Dream 2       TV         26  2007   
2340  11835    Qin Shi Ming Yue: Zhu Zi Bai Jia       TV         34  2010   
1527  36910               Wu Geng Ji 2nd Season      ONA         42  2017   
2375  15905  Qin Shi Ming Yue: Wanli Changcheng       TV         37  2012   
2990   2758               Shippuu! Iron Leaguer       TV         52  1993   
2165  36561            Doupo Cangqiong Specials  Special          2  2017   
2984  29936    Qin Shi Ming Yue: Junlin Tianxia      ONA         75  2014   
2548  38250                 Xiao Lu He Xiao Lan      ONA         72  2018   
2590  38491                      Xing Chen Bian      ONA         12  2018   

       ogledi  ocena  
rang                  
53    1582700    8.6  
113   1472352    8.4  
1410  1415367    7.5  
1     1327051    9.2  
45    1160132    8.7  
581   1123638    7.9  
3     1116166    9.1  
253   1089044    8.2  
177   1074522    8.3  
716   1073428    7.9  
...       ...    ...  
1941      781    7.4  
2963      769    7.1  
2340      720    7.3  
1527      715    7.5  
2375      707    7.3  
2990      707    7.1  
2165      697    7.3  
2984      501    7.1  
2548      437    7.2  
2590      374    7.2  

[2930 rows x 7 columns]

Iz te tabele pa lahko razberemo, da je do zdaj najbolj ogledan anime bil "Death Note" iz leta 2006, najmanj ogledov pa je prejel "Xing Chen Bian" iz 2018.

### IZRAČUNANA STATISTIKA
V tem delu naloge bom s pomočjo grafov odgovoril na zgoraj zastavljena vprašanja.
Najprej grupiramo animeje po desetletjih, da bodo grafi bolj pregledni, tam kjer bo to imelo kak efekt.

In [41]:
animeji['desetletje'] = 10 * (animeji.leto // 10)

In [42]:
animeji

id                                             naslov      tip  \
rang                                                                      
1      5114                   Fullmetal Alchemist: Brotherhood       TV   
2     32281                                     Kimi no Na wa.    Movie   
3      9253                                        Steins;Gate       TV   
4     28977                                           Gintama°       TV   
5     11061                             Hunter x Hunter (2011)       TV   
6       820                               Ginga Eiyuu Densetsu      OVA   
7      9969                                           Gintama'       TV   
8     35180                         3-gatsu no Lion 2nd Season       TV   
9     15417                                Gintama': Enchousen       TV   
10    28851                                     Koe no Katachi    Movie   
...     ...                                                ...      ...   
2991    478                                 Sousei no Aquarion       TV   
2992   2452                              Space Adventure Cobra    Movie   
2993  35774                         Starmyu 2nd Season Special  Special   
2994   3332                                 Stop!! Hibari-kun!       TV   
2995  21881                                Sword Art Online II       TV   
2996   4532                          Taiyou no Yuusha Fighbird       TV   
2997  20035  Toaru Majutsu no Index Movie: Endymion no Kise...    Movie   
2998   7654                          Tokyo Magnitude 8.0 Recap  Special   
2999   2577                               Tom Sawyer no Bouken       TV   
3000  31370                               Tonkatsu DJ Agetarou       TV   

      st_epizod  leto   ogledi  ocena  desetletje  
rang                                               
1            64  2009  1327051    9.2        2000  
2             1  2016   866945    9.1        2010  
3            24  2011  1116166    9.1        2010  
4            51  2015   226008    9.1        2010  
5           148  2011   818148    9.1        2010  
6           110  1988   140128    9.1        1980  
7            51  2011   235640    9.1        2010  
8            22  2017   114343    9.0        2010  
9            13  2012   137790    9.0        2010  
10            1  2016   621825    9.0        2010  
...         ...   ...      ...    ...         ...  
2991         26  2005    57573    7.1        2000  
2992          1  1982     5258    7.1        1980  
2993          1  2017     1649    7.1        2010  
2994         35  1983     2996    7.1        1980  
2995         24  2014   911220    7.1        2010  
2996         48  1991     1746    7.1        1990  
2997          1  2013    15645    7.1        2010  
2998          1  2009     5668    7.1        2000  
2999         49  1980     3648    7.1        1980  
3000         12  2016     9884    7.1        2010  

[2930 rows x 8 columns]

Sedaj se lotimo prvega vprašanja: Ali je zlata doba animejev že mimo, ali se šele začenja?

In [47]:
animeji[animeji.ocena >= 9.0].plot.scatter('desetletje', 'ocena')